jgrapht-core/src/main/java/org/jgrapht/alg/lca/HeavyPathLCAFinder.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.lca;
+
+import org.jgrapht.Graph;
+import org.jgrapht.alg.decomposition.HeavyPathDecomposition;
+import org.jgrapht.alg.interfaces.LCAAlgorithm;
+
+import java.util.*;
+
+/**
+ * Algorithm for computing lowest common ancestors in rooted trees and forests based on {@link HeavyPathDecomposition}.
+ *
```

Comment:
Add some reference to appropriate paper which introduced the technique. If this is not explicit to a particular paper, add a reference to the dynamic trees paper of Sleator and Tarjan which made all these possible.